app/controllers/step_elements/checklist_items_controller.rb

```diff
@@ -35,17 +41,31 @@ def update
       @checklist_item.assign_attributes(
         checklist_item_params.except(:position, :id).merge(last_modified_by: current_user)
       )
-
-      if @checklist_item.save!
-        log_activity(
-          "#{@step.protocol.in_module? ? :task : :protocol}_step_checklist_item_edited",
-          checklist_item: @checklist_item.text,
-          checklist_name: @checklist.name
-        )
-        checklist_item_annotation(@step, @checklist_item, old_text)
+      new_items = []
+      ActiveRecord::Base.transaction do
+        new_items = @checklist_item.save_multiline!
+        new_items.each_with_index do |item, i|
+          if i.zero?
+            log_activity(
+              "#{@step.protocol.in_module? ? :task : :protocol}_step_checklist_item_edited",
+              checklist_item: item.text,
+              checklist_name: @checklist.name
+            )
+            checklist_item_annotation(@step, item, old_text)
+          else
+            log_activity(
+              "#{@step.protocol.in_module? ? :task : :protocol}_step_checklist_item_added",
+              {
+                checklist_item: item.text,
+                checklist_name: @checklist.name
+              }
+            )
+            checklist_item_annotation(@step, item)
```

Comment:
Instead of the if block we could just do:
`"#{@step.protocol.in_module? ? :task : :protocol}_step_checklist_item_#{i.zero? 'edited' : 'added'}",`
and then
` checklist_item_annotation(@step, item) unless i.zero?`